### ISSUE TITLE
Dropdowns broken on iOS mobile/tablet

### ISSUE THREAD

**Issue by stevokk** [angular-ui/bootstrap#3242]
I was experiencing an issue on 0.12.0 & 0.11.2 where the dropdowns were not closing when clicking outside the element unless I clicked the toggle element directly.

Before fix: All desktops and kindle fire worked, ipad mini and air did not
After fix: All desktops, kindle, and ipads worked

This happens even on the examples page:
http://angular-ui.github.io/bootstrap/#/dropdown

This issue is that on iOS mobile devices the following lines are not triggered:
$document.bind('click', closeDropdown);
$document.unbind('click', closeDropdown);

These reside here:
https://github.com/angular-ui/bootstrap/blob/master/src/dropdown/dropdown.js Line 12 & 26

Fix:
$document.bind('click touchstart', closeDropdown);
$document.unbind('click touchstart', closeDropdown);

Reason explained in this thread:
http://stackoverflow.com/questions/7892863/jquery-click-not-working-with-ipad
tl:dr iOS doesn't fire events for elements that do not have an event directly attached to them unless they are link node types.

Unfortunately I am unable to create a pull request due to company policy.

Best Regards,
Steven

**Comment by stevokk** [comment]
After more testing I realised that touchstart breaks the contents if it contains links, another workaround for this is to set cursor:pointer css for iOS only. Not sure if this falls under a library problem but should probably be covered.

**Comment by mappmechanic** [comment]
In iOS, it should trigger native dropdown.

**Comment by stevokk** [comment]
Why would a custom dropdown element trigger a native select box? I think you're mixing selects and dropdowns. Up to a bootstrap developer to close this issue but I believe the framework should account for the use of iOS browsers. My quick fix was the following:
```
  // Fix iOS click event dispatching for bootstrap dropdown
  angular.element(window).bind('load', function () {
    var iOS = (navigator.userAgent.match(/(iPad|iPhone|iPod)/g) ? true : false);
    if (iOS) {
      angular.element(document.body).css('cursor', 'pointer');
    }
  });
```

**Comment by jsdevel** [comment]
@stevokk thanks for providing the fix!  I verified it locally and issued #3257 to get this resolved.  Hopefully it's handled soon.

**Comment by alexmarh** [comment]
@stevokk thanks a lot, it was very helpful for me

**Comment by wesleycho** [comment]
Not able to reproduce this currently - closing as already resolved.
